File: Recursion/factorial.cpp
#include <iostream>

using namespace std;

int factorial(int n)
{
    if (n > 0)
    {
        return factorial(n - 1) * n;
    }
    else
    {
        return 1;
    }
}

int main()
{
    cout << factorial(5);
    return 0;
}

File: Mathematics/Factorial.java
package Mathematics;

public class Factorial {
    static int loop(int n) {
        int res = 1;
        for (int i = 2; i <= n; i++) {
            res = res * i;
        }
        return res;
    }

    static int recursion(int n) {
        if (n == 0) {
            return 1;
        } else {
            return n * recursion(n - 1);
        }
    }

    public static void main(String args[]) {
        int num = 4;

        System.out.println("Loop " + loop(num));
        System.out.println("Recursion " + recursion(num));
    }
}
